plugins/woocommerce/client/admin/client/wp-admin-scripts/fulfillments/utils/fulfillment-utils.ts

```diff
@@ -28,6 +35,67 @@ export function getFulfillmentItems(
 	) as Array< FulfillmentItem >;
 }
 
+export function hasPendingItems(
+	order: Order,
+	fulfillments: Fulfillment[]
+): boolean {
+	return order.line_items.some( ( item: LineItem ) => {
+		const totalFulfilledQty = fulfillments.reduce(
+			( total, fulfillment ) => {
+				const fulfillmentItems = getFulfillmentItems( fulfillment );
+				const fulfillmentItemInOrder = fulfillmentItems.find(
+					( fItem: FulfillmentItem ) => fItem.item_id === item.id
+				);
+				return total + ( fulfillmentItemInOrder?.qty || 0 );
+			},
+			0
+		);
+
+		return totalFulfilledQty < item.quantity;
+	} );
+}
+
+export async function refreshOrderFulfillmentStatus( orderId: number ) {
+	dispatch( FulfillmentStore ).invalidateResolution( 'getOrder', [
+		orderId,
+	] );
+	const order: Order | null = await resolveSelect(
+		FulfillmentStore
+	).getOrder( orderId );
+	if ( order ) {
+		const order_status =
+			( order.meta_data.find(
+				( meta ) => meta.key === '_fulfillment_status'
+			)?.value as string ) ?? 'no_fulfillments';
+		const marker = document.querySelector(
+			`.order-${ orderId } td.fulfillment_status mark`
+		);
+		if ( marker ) {
+			const status = window.wcFulfillmentSettings
+				.order_fulfillment_statuses[ order_status ] || {
+				label: __( 'Unknown', 'woocommerce' ),
+				background_color: '#f8f9fa',
+				text_color: '#6c757d',
+			};
+			// Set content of the marker to the label of the status.
+			const textContainer = marker.querySelector( 'span' );
+			if ( textContainer ) {
+				textContainer.textContent = status.label;
```

Comment:
should we sanitize `status.label`?

Reply:
Does it have a chance to be misused? We don't send back the value, and it is coming from the server as sanitized. If someone alters it, it will stay on their browser.